### PR TITLE
replace {account_id} and  {region} in policy mode for multiaccount sc…

### PR DESCRIPTION
**Pull request by cliffpracht**
…enarios

Addresses this issue:

https://github.com/capitalone/cloud-custodian/issues/1882

### CODE REVIEW

**Comment by thisisshi** [comment]
account_id and region interpolation is handled in c7n.resolver currently, better to reuse the code from there: https://github.com/capitalone/cloud-custodian/blob/master/c7n/resolver.py

**Comment by cliffpracht** [comment]
Ok thanks @thisisshi.  I will take a look at the c7n.resolver.  I was following some of the work that @FireballDWF had done for `mode['role']`

**Comment by kapilt** [comment]
@cliffpracht thanks for the pr, would you mind signing the contributor license agreement per the project readme, direct link https://docs.google.com/forms/d/19LpBBjykHPox18vrZvBbZUcK6gQTj7qv1O5hCduAZFU/viewform

**Comment by kapilt** [comment]
@cliffpracht expand_variables on the policy is the right place, but switching out to using utils.format_string_values will handle expansion a bit more nicely, c7n.resolver.ValuesFrom.__init__ as an example of that.

**Comment by cliffpracht** [comment]
Got it - I'll take a look at c7n.resolver.ValuesFrom.init as an example.  I also signed the contributor license agreement.

**Comment by cliffpracht** [comment]
@kapilt, @thisisshi,  I have updated this to use the `utils.format_string_values` method.

**Comment by kapilt** [comment]
thanks, looks good, i'm not sure if we should do the whole policy or just the 'mode' section.. when we do the whole policy we start getting into implicit semantics that are hard to walk back.. like a policy filter for a value of "{xyz}" and getting a key error because of variable expansion.

**Comment by kapilt** [comment]
we'll probably need a follow up branch to make this more wide spread, as its only going to expand policy vars/interpolation as things go to lambda provisioning.

File: c7n/policy.py [hunk 1]
```diff
@@ -465,9 +454,10 @@ def provision(self):
                 "Provisioning policy lambda %s", self.policy.name)
             variables = {
                 'account_id': self.policy.options.account_id,
-                'policy': self.policy.data
+                'policy': self.policy.data,
+                'region': self.policy.options.region or 'default'
```
Comment (kapilt) [inline c7n/policy.py]: not sure what the or 'default' bit is about, there isn't really a default value here. i removed it and tests pass fine.

Comment (cliffpracht) [reply c7n/policy.py]: I was following the lead of this bit of code [here](https://github.com/capitalone/cloud-custodian/blob/db7c4718f2fdde54e0559869e4149b2725061747/c7n/policy.py#L225). 

So if `self.policy.options.region` always has a value then I can remove it.

**Comment by cliffpracht** [comment]
Agree on follow up branch to make widespread.  
As I understand(although I have to admit I am fairly new to this codebase), there would potentially be a need to interpolate the policy at run time versus at deploy time in the case of assume role for instance.   Where you may want to use the region or account of the role you are assuming instead of the region/account that you would be deploying to.

**Comment by kapilt** [comment]
so till we make it more wide spread, i think we just keep to this to mode part, otherwise its a differential on behavior compared to other execution modes (mainly the default cli pull mode). i'll file a separate issue for wider spread interpolation on policies (we probably want to handle that primarily in validate to catch any issues prior to execution). wrt to the interpolate at runtime / deploy time, we're already assumed by the time we get to policy execution/provisioning. the only case provisioning and execution would really be different is if we're using a single lambda across accounts, which we do support in some limited cases today with intent to grow capabilities around.

**Comment by cliffpracht** [comment]
Totally agree, @kapilt.  I'll change up my PR to only address the mode part.

File: c7n/policy.py (hunk 1 above)
Comment (kapilt) [reply c7n/policy.py]: yeah.. it should always have a value, i see what you mean though, i think that default there was for testing, in practice when executing we're always targeting a known region which is set before we get to this code.

**Comment by kapilt** [comment]
filed #2330 for adding this more widespread.

**Pull request merged**